docs/rules/idiomatic/non-raw-regex-pattern.md

```diff
@@ -0,0 +1,60 @@
+# non-raw-regex-pattern
+
+**Summary**: Use raw strings for regex patterns
+
+**Category**: Idiomatic
+
+**Avoid**
+```rego
+all_digits if {
+    regex.match("[\\d]+", "12345")
+}
+```
+
+**Prefer**
+```rego
+all_digits if {
+    regex.match(`[\d]+`, "12345")
+}
+```
+
+## Rationale
+
+[Raw strings](https://www.openpolicyagent.org/docs/edge/policy-language/#strings) are interpreted literally, allowing
+you to avoid having to escape special characters like `\` in your regex patterns. Using raw strings for regex patterns
+additionally makes them easier to identify as such.
+
+## Limitations
+
+This rule currently only scans regex string literals in the place of the `pattern` argument of the various
+[regex built-in functions](https://www.openpolicyagent.org/docs/latest/policy-reference/#regex). It will not **not**
+try to "resolve" patterns assigned to variables. The following example would as such not render a warning:
+
+```rego
+package policy
+
+import future.keywords.if
+
+# Pattern assigned to variable
+pattern := "[\\d]+"
+
+# This won't trigger a violation
+allow if regex.match(pattern, "12345")
+```
```

Comment:
is it worth opening an issue for this too?

Reply:
Yeah, not sure about this specifically, but we could definitely add a feature request for being able to track values in scope. I don't think it can be easily done in Rego alone, at least not for more advanced cases where the var could be assigned the value of another var, which could be assigned the value of another var, etc... but perhaps a simpler solution would be good enough for most use cases.

Reply:
https://github.com/StyraInc/regal/issues/229